robocop/run.py

```diff
@@ -32,6 +33,26 @@ def __init__(self, from_cli=False):
         self.load_reports()
         self.configure_checkers_or_reports()
 
+    def export_results(self, *issues):
+        """ Gather results from issues and reports, merge them in a list and return the data"""
+        payload = []
+        for report in self.reports:
+            output = report.get_report()
+            if output is not None:
+                payload.append(output)
+        for issue in self.issues:
+            issue = {
+                "source": issue.source,
```

Comment:
You can add serializing method to Message class so it would return dict:
```
class Message:
    (...)
    def to_json(self):
        return {
            "source": self.source,
            # and so on
        }
```